modular/gater/object_handler.go

```
@@ -100,6 +100,14 @@ func (g *GateModular) putObjectHandler(w http.ResponseWriter, r *http.Request) {
 		err = ErrInvalidPayloadSize
 		return
 	}
+	taskState, _, _ := g.baseApp.GfSpClient().GetUploadObjectState(reqCtx.Context(), objectInfo.Id.Uint64())
+	if taskState == int32(servicetypes.TaskState_TASK_STATE_UPLOAD_OBJECT_DONE) {
+		// It is not allowed to upload piece or object for an object id which had already been fully uploaded.
+		log.CtxErrorw(reqCtx.Context(), "failed to put object as the target object had already fully uploaded")
+		err = ErrInvalidUploadRequest
+		return
+	}
```

Move to uploader.PreUploadObject may be better.

---

Ignoring errors may lead to the wrong branch under exception.